src/main.cpp

```diff
@@ -49,16 +50,31 @@ int main(int argc, char **argv){
     // Create jointstate publisher (for use with state_publisher node)
     ros::NodeHandle nodehandle;
     ros::Publisher JointPublisher = nodehandle.advertise<sensor_msgs::JointState>("joint_states", 10, false);
-    double originX;
-    double originY;
-    {
-        cv::Mat image = cv::imread(heightmap_filename, cv::IMREAD_GRAYSCALE);
-        originX = (image.rows * 0.025 / 2);
-        originY = -(image.cols * 0.025 / 2);
-    }
 
     // Create location broadcaster
     tf2_ros::TransformBroadcaster LocationBroadcaster;
+    tf2_ros::StaticTransformBroadcaster OriginBroadcaster;
+
+    {
+        cv::Mat image = cv::imread(heightmap_filename, cv::IMREAD_GRAYSCALE);
+        double originX = (image.cols * 0.025 / 2);
+        double originY = (image.rows * 0.025 / 2);
+        geometry_msgs::TransformStamped OriginMsg;
+        OriginMsg.header.stamp = ros::Time(0);
+        OriginMsg.header.frame_id = "map";
+        OriginMsg.child_frame_id = "origin";
+        OriginMsg.transform.translation.x = originX;
+        OriginMsg.transform.translation.y = originY;
+        OriginMsg.transform.translation.z = 0;
+        tf2::Quaternion q;
+        q.setRPY(0, 0, 1.570796);
+        OriginMsg.transform.rotation.x = q.x();
+        OriginMsg.transform.rotation.y = q.y();
+        OriginMsg.transform.rotation.z = q.z();
+        OriginMsg.transform.rotation.w = q.w();
+
+        OriginBroadcaster.sendTransform(OriginMsg);
+    }
```

Comment:
Why do you need an additional frame? When using the `map_server`, you can define the origin of the map. So you an place the `map` frame anywhere you like, including rotation.

See http://wiki.ros.org/map_server#YAML_format